api/parts/data/fetch.js

```diff
@@ -1827,4 +1829,100 @@ function union(x, y) {
     return res;
 }
 
+/**
+* Get data for jobs listing for jobs api
+* @param {string} metric - name of the collection where to get data from
+* @param {params} params - params object with app_id and date
+*/
+fetch.fetchJobs = async function(metric, params) {
+    try {
+        if (params.qstring.name) {
+            await fetch.jobDetails(metric, params);
+        }
+        await fetch.alljobs(metric, params);
+    }
+    catch (e) {
+        console.log(e);
+        common.returnOutput(params, 500, "Fetching jobs failed");
+    }
+};
+
+/**
+* Get all jobs grouped by job name for jobs api
+* @param {string} metric - name of the collection where to get data from
+* @param {params} params - params object with app_id and date
+*/
+fetch.alljobs = async function(metric, params) {
+    const columns = ["name", "schedule", "next", "finished", "status", "total"];
+    let sort = {};
+    let total = await countlyDb._native.collection('jobs').aggregate([
+        {
+            $group: { _id: "$name" }
+        },
+        {
+            $count: 'total'
+        }
+    ]).toArray();
+    total = total.length > 0 ? total[0].total : 0;
+    const pipeline = [
+        {
+            $sort: {
+                finished: -1
```

Comment:
the sort is usually the first one that can exhaust pipeline step memory limit, so it makes sense to add indexes for such fields.
We can simply call create an index on process start, like here:
https://github.com/Countly/countly-server/blob/master/frontend/express/app.js#L1539-L1541

Reply:
updated! added index.